### ISSUE TITLE
jsThis can only be called within a delegate

### ISSUE THREAD

**Issue by Neftedollar** [fable-compiler/Fable#633]
### Description

I've catch this error:

[ERROR] `jsThis` can only be called within a delegate (L21,56-L21,62) (/home/neftedollar/Documents/feteor/feteor/ExampleApplication/fsx_example/main.fsx)

but I didn't touch that code.

### Repro code

```fs
Meteor.startup (fun _ -> 
    printfn "Meteor.isServer %A" Meteor.isServer
    printfn "Meteor.isClient %A" Meteor.isClient
    printfn "Meteor.isCordova %A" Meteor.isCordova
    printfn "Meteor.isDevelopment %A" Meteor.isDevelopment
    printfn "Meteor.isProduction %A" Meteor.isProduction
    printfn "Meteor.isTest %A" Meteor.isTest
    printfn "Meteor.release %A" Meteor.release
    printfn "Meteor.settings %A" Meteor.settings

    let methods = createObj [
        "new_method" ==> JsFunc0(fun () -> printfn "%A" jsThis?userId) //this!
        "anotherTestMethod" ==> JsFunc1(fun a -> printfn "%A" a)
    ]
    Meteor.methods methods
    )
```

### Expected and actual results

It should compile but It don't.

### Related information 

* fable-compiler version: 0.7.22
* fable-core version: 0.7.15
* fable-powerpack: 0.0.19
* Operating system: Ubuntu 16.04

npm-log
```
0 info it worked if it ends with ok
1 verbose cli [ '/home/neftedollar/.meteor/packages/meteor-tool/.1.4.2_3.1n7sxp8++os.linux.x86_64+web.browser+web.cordova/mt-os.linux.x86_64/dev_bundle/bin/node',
1 verbose cli   '/home/neftedollar/.meteor/packages/meteor-tool/.1.4.2_3.1n7sxp8++os.linux.x86_64+web.browser+web.cordova/mt-os.linux.x86_64/dev_bundle/bin/npm',
1 verbose cli   'run',
1 verbose cli   'fable' ]
2 info using npm@3.10.9
3 info using node@v4.6.2
4 verbose run-script [ 'prefable', 'fable', 'postfable' ]
5 info lifecycle fsx_example@~prefable: fsx_example@
6 silly lifecycle fsx_example@~prefable: no script for prefable, continuing
7 info lifecycle fsx_example@~fable: fsx_example@
8 verbose lifecycle fsx_example@~fable: unsafe-perm in lifecycle true
9 verbose lifecycle fsx_example@~fable: PATH: /home/neftedollar/.meteor/packages/meteor-tool/.1.4.2_3.1n7sxp8++os.linux.x86_64+web.browser+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/npm/bin/node-gyp-bin:/home/neftedollar/Documents/feteor/feteor/ExampleApplication/fsx_example/node_modules/.bin:/home/neftedollar/.meteor/packages/meteor-tool/.1.4.2_3.1n7sxp8++os.linux.x86_64+web.browser+web.cordova/mt-os.linux.x86_64/dev_bundle/bin:/home/neftedollar/.meteor/packages/meteor-tool/.1.4.2_3.1n7sxp8++os.linux.x86_64+web.browser+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/.bin:/home/neftedollar/bin:/home/neftedollar/.local/bin:/home/neftedollar/bin:/home/neftedollar/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/usr/local/share/jdk1.8.0_111/bin:/usr/local/share/scala-2.11.8/bin/:/snap/bin:/usr/local/share/jdk1.8.0_111/bin:/usr/local/share/scala-2.11.8/bin/
10 verbose lifecycle fsx_example@~fable: CWD: /home/neftedollar/Documents/feteor/feteor/ExampleApplication/fsx_example
11 silly lifecycle fsx_example@~fable: Args: [ '-c', 'fable' ]
12 silly lifecycle fsx_example@~fable: Returned: code: 1  signal: null
13 info lifecycle fsx_example@~fable: Failed to exec fable script
14 verbose stack Error: fsx_example@ fable: `fable`
14 verbose stack Exit status 1
14 verbose stack     at EventEmitter.<anonymous> (/home/neftedollar/.meteor/packages/meteor-tool/.1.4.2_3.1n7sxp8++os.linux.x86_64+web.browser+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/npm/lib/utils/lifecycle.js:255:16)
14 verbose stack     at emitTwo (events.js:87:13)
14 verbose stack     at EventEmitter.emit (events.js:172:7)
14 verbose stack     at ChildProcess.<anonymous> (/home/neftedollar/.meteor/packages/meteor-tool/.1.4.2_3.1n7sxp8++os.linux.x86_64+web.browser+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/npm/lib/utils/spawn.js:40:14)
14 verbose stack     at emitTwo (events.js:87:13)
14 verbose stack     at ChildProcess.emit (events.js:172:7)
14 verbose stack     at maybeClose (internal/child_process.js:854:16)
14 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:222:5)
15 verbose pkgid fsx_example@
16 verbose cwd /home/neftedollar/Documents/feteor/feteor/ExampleApplication/fsx_example
17 error Linux 4.4.0-57-generic
18 error argv "/home/neftedollar/.meteor/packages/meteor-tool/.1.4.2_3.1n7sxp8++os.linux.x86_64+web.browser+web.cordova/mt-os.linux.x86_64/dev_bundle/bin/node" "/home/neftedollar/.meteor/packages/meteor-tool/.1.4.2_3.1n7sxp8++os.linux.x86_64+web.browser+web.cordova/mt-os.linux.x86_64/dev_bundle/bin/npm" "run" "fable"
19 error node v4.6.2
20 error npm  v3.10.9
21 error code ELIFECYCLE
22 error fsx_example@ fable: `fable`
22 error Exit status 1
23 error Failed at the fsx_example@ fable script 'fable'.
23 error Make sure you have the latest version of node.js and npm installed.
23 error If you do, this is most likely a problem with the fsx_example package,
23 error not with npm itself.
23 error Tell the author that this fails on your system:
23 error     fable
23 error You can get information on how to open an issue for this project with:
23 error     npm bugs fsx_example
23 error Or if that isn't available, you can get their info via:
23 error     npm owner ls fsx_example
23 error There is likely additional logging output above.
24 verbose exit [ 1, true ]

```

**Comment by Neftedollar** [comment]
I've changed binding file
from 
```fs
        static member defer(func: System.Action): unit = jsNative
        static member startup(func: System.Action): unit = jsNative
```

to

```fs
        [<Emit("Meteor.defer($0)")>]
        static member defer(func: unit -> unit): unit = jsNative
        [<Emit("Meteor.startup($0)")>]
        static member startup(func: unit -> unit): unit = jsNative
```

and this is code that uses that binding
```fs
Meteor.startup (fun _ -> ///!!!
    printfn "Meteor.isServer %A" Meteor.isServer
    printfn "Meteor.isClient %A" Meteor.isClient
    printfn "Meteor.isCordova %A" Meteor.isCordova
    printfn "Meteor.isDevelopment %A" Meteor.isDevelopment
    printfn "Meteor.isProduction %A" Meteor.isProduction
    printfn "Meteor.isTest %A" Meteor.isTest
    printfn "Meteor.release %A" Meteor.release
    printfn "Meteor.settings %A" Meteor.settings

    let methods = createObj [
        "new_method" ==> JsFunc0(fun () -> printfn "%A" jsThis?userId) //this!
        "anotherTestMethod" ==> JsFunc1(fun a -> printfn "%A" a)
    ]
    Meteor.methods methods
    )
```
```fs
Meteor.defer(fun () -> 
    if Meteor.isServer then
        printfn "%A" Meteor.methods
    Meteor.call("new_method") |> ignore
    Meteor.call("anotherTestMethod", "lol") |> ignore
    Meteor.call("anotherTestMethod", 3) |> ignore
    Meteor.call("anotherTestMethod", createObj [ "lol" ==> 3 ]) |> ignore
)

```

**Comment by Neftedollar** [comment]
@alfonsogarciacaro thank you.

**Comment by alfonsogarciacaro** [comment]
Sorry, this was a silly mistake on my side. Should be fixed in fable-compiler 0.7.26 👍

**Comment by Neftedollar** [comment]
It's ok. Thanks again!
